linode/manager
# feat: [M3-6759] - Handle VPC Account Grants / Permissions / Capabilities

## Description 📝
Add VPC grants, permissions, and capabilities to determine VPC usage

## Major Changes 🔄
- add VPC related grants to User Permissions page
- add isFeatureEnabledFunctionality

## Preview 📷

User Permissions:

![image](https://github.com/linode/manager/assets/139280159/b2af10ed-416e-4c9c-8f09-2f242875af35)
no permissions:
![image](https://github.com/linode/manager/assets/139280159/1b7fbe32-fef6-4f39-b762-e3cec7b4f4fa)
![image](https://github.com/linode/manager/assets/139280159/1e4e4099-c9dc-4800-8024-f35df5fbc6b0)
![image](https://github.com/linode/manager/assets/139280159/a3966e88-5a05-4aa5-9356-1c9e4574f954)

| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/linode/manager/assets/139280159/fc4db107-26ae-4652-9150-3692e66238c8) | ![image](https://github.com/linode/manager/assets/139280159/3c1e3836-9abb-4c32-a553-d1b3da705fc6) |

## How to test 🧪
1. Test that user permissions ^ show up (either on MSW or possibly even dev account?)
2. Test if you can see any VPC pages while not on MSW (will need to have the relevant capability)
3. try to test editing a VPC when it is read_only (may need to add another user to change their permissions)
4. Try creating a vpc on an account that does not have the global permissions
